lightning/src/ln/channelmanager_limits_tests.rs

```diff
@@ -673,8 +673,8 @@ fn test_channel_update_cached() {
 
 	{
 		// Assert that ChannelUpdate message has been added to node[0] pending broadcast messages
-		let pending_broadcast_messages= nodes[0].node.pending_broadcast_messages.lock().unwrap();
-		assert_eq!(pending_broadcast_messages.len(), 1);
+		let pending_broadcast_messages = nodes[0].node.get_and_clear_pending_msg_events();
+		assert_eq!(pending_broadcast_messages.len(), 2);
```

Comment:
@TheBlueMatt also refactoring the `pending_broadcast_message` to `get_and_clear_pending_msg_events` is affecting the check here ( https://github.com/srikanth-iyengar/rust-lightning/blob/f585d357a4bb5126bf3c379bfd8a0dfa82470997/lightning/src/ln/channelmanager_limits_tests.rs#L698-L701 ) to fail

Reply:
This test was introduced in #2731

Reply:
https://github.com/lightningdevkit/rust-lightning/pull/2977/commits/f585d357a4bb5126bf3c379bfd8a0dfa82470997

it will be better to take a look at this diff